utils/serverHelpers.js

```
@@ -0,0 +1,242 @@
+      return callback(null, true);
+    }
+    
+    // Check exact matches first
+    if (allowedOrigins.includes(origin)) {
+      console.log('✅ CORS: Origin allowed (exact match):', origin);
+      return callback(null, true);
+    }
+    
+    // Check PR preview pattern
+    if (prPreviewPattern && prPreviewPattern.test(origin)) {
+      console.log('✅ CORS: Origin allowed (PR preview):', origin);
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)